src/dynamic_win_raw.rs

```diff
@@ -0,0 +1,407 @@
+type PcapFileno = unsafe extern "C" fn(arg1: *mut pcap_t) -> c_int;
+type PcapDumpOpen =
+    unsafe extern "C" fn(arg1: *mut pcap_t, arg2: *const c_char) -> *mut pcap_dumper_t;
+type PcapDumpFopen = unsafe extern "C" fn(arg1: *mut pcap_t, fp: *mut FILE) -> *mut pcap_dumper_t;
+type PcapDumpFlush = unsafe extern "C" fn(arg1: *mut pcap_dumper_t) -> c_int;
+type PcapDumpClose = unsafe extern "C" fn(arg1: *mut pcap_dumper_t);
+type PcapDump =
+    unsafe extern "C" fn(arg1: *mut c_uchar, arg2: *const pcap_pkthdr, arg3: *const c_uchar);
+type PcapFindalldevs = unsafe extern "C" fn(arg1: *mut *mut pcap_if_t, arg2: *mut c_char) -> c_int;
+type PcapFreealldevs = unsafe extern "C" fn(arg1: *mut pcap_if_t);
+type PcapGetSelectableFd = unsafe extern "C" fn(arg1: *mut pcap_t) -> c_int;
+type PcapSetTstampType = unsafe extern "C" fn(arg1: *mut pcap_t, arg2: c_int) -> c_int;
+type PcapFopenOfflineWithTstampPrecision =
+    unsafe extern "C" fn(arg1: *mut FILE, arg2: c_uint, arg3: *mut c_char) -> *mut pcap_t;
+type PcapOpenDeadWithTstampPrecision =
+    unsafe extern "C" fn(arg1: c_int, arg2: c_int, arg3: c_uint) -> *mut pcap_t;
+type PcapOpenOfflineWithTstampPrecision =
+    unsafe extern "C" fn(arg1: *const c_char, arg2: c_uint, arg3: *mut c_char) -> *mut pcap_t;
+type PcapSetImmediateMode = unsafe extern "C" fn(arg1: *mut pcap_t, arg2: c_int) -> c_int;
+type PcapSetTstampPrecision = unsafe extern "C" fn(arg1: *mut pcap_t, arg2: c_int) -> c_int;
+type PcapDumpOpenAppend =
+    unsafe extern "C" fn(arg1: *mut pcap_t, arg2: *const c_char) -> *mut pcap_dumper_t;
+type PcapSetmintocopy = unsafe extern "C" fn(arg1: *mut pcap_t, arg2: c_int) -> c_int;
+type PcapGetEvent = unsafe extern "C" fn(p: *mut pcap_t) -> HANDLE;
+type PcapSendqueueAlloc = unsafe extern "C" fn(memsize: c_uint) -> *mut pcap_send_queue;
+type PcapSendqueueDestroy = unsafe extern "C" fn(queue: *mut pcap_send_queue);
+type PcapSendqueueQueue = unsafe extern "C" fn(
+    queue: *mut pcap_send_queue,
+    pkt_header: *const pcap_pkthdr,
+    pkt_data: *const c_uchar,
+) -> c_int;
+type PcapSendqueueTransmit =
+    unsafe extern "C" fn(p: *mut pcap_t, queue: *mut pcap_send_queue, sync: c_int) -> c_uint;
+
+static mut LIBRARY: Lazy<Library> = Lazy::new(|| unsafe { load_library() });
+
+unsafe fn load_library() -> Library {
+    let mut libfile = PathBuf::from("wpcap.dll");
+
+    let libdirpath = if let Ok(libdir) = env::var("LIBPCAP_LIBDIR") {
+        Some(PathBuf::from(&libdir))
+    } else {
+        None
+    };
+
+    if let Some(libdir) = libdirpath {
+        libfile = libdir.join(libfile);
+    }
+
+    Library::new(libfile).unwrap()
+}
+
+pub unsafe fn pcap_create(arg1: *const c_char, arg2: *mut c_char) -> *mut pcap_t {
+    let func = LIBRARY.get::<PcapCreate>(b"pcap_create").unwrap();
+    func(arg1, arg2)
+}
+pub unsafe fn pcap_set_snaplen(arg1: *mut pcap_t, arg2: c_int) -> c_int {
+    let func = LIBRARY.get::<PcapSetSnaplen>(b"pcap_set_snaplen").unwrap();
+    func(arg1, arg2)
+}
```

Comment:
It does panic, however since our consumer is in its own thread, it only panics that thread.
A more refined solution should change the function signatures